hastory-data/src/Data/Hastory/Types.hs

```diff
@@ -44,3 +49,14 @@ instance Hashable Entry where
 
 instance Validity Entry where
     isValid Entry {..} = isValid entryText && isValid entryWorkingDir
+
+instance ToJSON Entry
+instance FromJSON Entry
+
+data EntryWithKey = EntryWithKey
+  { _ewkKey   :: Int64
```

Comment:
These won't be globally unique. When multiple clients all send entry number 0, we'll run into trouble.
For now, forget about deduplication, and forget about this key :)

Reply:
Removed logic related to deduplication, then.